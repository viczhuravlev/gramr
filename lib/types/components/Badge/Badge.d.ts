/// <reference types="react" />
import * as T from './Badge.types';
declare function Badge(props: T.BadgeProps): JSX.Element;
declare namespace Badge {
    var defaultProps: Pick<import("..").TextProps, "size">;
}
export default Badge;
