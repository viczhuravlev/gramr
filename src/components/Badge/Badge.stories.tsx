import React from 'react';
import { css } from 'linaria';
import { styled } from 'linaria/react';
import { storiesOf } from '@storybook/react';

import { sizes } from '../../styles/common';

import BadgeOrigin from './index';

const Badge = styled(BadgeOrigin)`
  & + & {
    margin-left: 8px;
  }
`;

const NewBadge = styled(Badge)`
  background-color: #536390;
`;

const newStyle = css`
  color: var(--gramr-ui-font-color);

  background-color: transparent;
  box-shadow: 0 0 0 1px var(--gramr-ui-font-color) inset;
`;

storiesOf('Badge', module)
  .add('Default', () => (
    <>
      <Badge>5</Badge>
      <Badge>99+</Badge>
      <Badge>999+</Badge>
      <Badge>9999+</Badge>
      <Badge>99999+</Badge>
      <Badge>Badge</Badge>
      <Badge>Badge with big text</Badge>
    </>
  ))
  .add('Size', () => (
    <>
      {sizes.map((size) => (
        <Badge size={size}>{size}</Badge>
      ))}
    </>
  ))
  .add('Custom style', () => (
    <>
      <Badge>Default</Badge>
      <NewBadge>Styled</NewBadge>
      <Badge className={newStyle}>New style</Badge>
    </>
  ));
