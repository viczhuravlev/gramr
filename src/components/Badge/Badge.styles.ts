import { styled } from 'linaria/react';

import { Text } from '../Text';

export const Badge = styled(Text)`
  display: inline-flex;

  align-items: center;

  justify-content: center;

  min-width: 28px;
  height: 28px;
  padding: 0 9px;

  color: #fff;

  background-color: var(--gramr-prominent-color);
  border-radius: 14px;

  &[data-size='xs'] {
    min-width: 20px;
    height: 20px;

    padding: 0 6px;

    border-radius: 10px;
  }

  &[data-size='s'] {
    min-width: 24px;
    height: 24px;

    padding: 0 8px;

    border-radius: 12px;
  }

  &[data-size='m'] {
    min-width: 28px;
    height: 28px;

    padding: 0 9px;

    border-radius: 14px;
  }

  &[data-size='l'] {
    min-width: 32px;
    height: 32px;

    padding: 0 10px;

    border-radius: 16px;
  }

  &[data-size='xl'] {
    min-width: 36px;
    height: 36px;

    padding: 0 12px;

    border-radius: 18px;
  }
`;
