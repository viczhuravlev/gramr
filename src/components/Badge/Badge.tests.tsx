import React from 'react';
import renderer from 'react-test-renderer';

import Badge from './index';

describe('[component] Badge', () => {
  test('[Snapshot] Badge', () => {
    const component = renderer.create(
      <Badge>
        Steve fish muskellunge swallower triplefin blenny horsefish wolffish.
      </Badge>
    );

    expect(component).toMatchSnapshot();
  });
});
