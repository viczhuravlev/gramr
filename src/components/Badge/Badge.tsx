import React from 'react';

import * as T from './Badge.types';
import * as S from './Badge.styles';

function Badge(props: T.BadgeProps): JSX.Element {
  const { style, children, className, ...otherProps } = props;

  return (
    <S.Badge style={style} className={className} {...otherProps}>
      {children}
    </S.Badge>
  );
}

const defaultProps: T.BadgeDefaultProps = {
  size: 'm',
};

Badge.defaultProps = defaultProps;

export default Badge;
