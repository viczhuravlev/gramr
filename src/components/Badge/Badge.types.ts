import { ReactNode, CSSProperties } from 'react';

import { TextProps } from '../Text';

export type BadgeDefaultProps = Pick<TextProps, 'size'>;

export interface BadgeProps extends Partial<BadgeDefaultProps> {
  style?: CSSProperties;
  children: ReactNode;
  className?: string;
}
