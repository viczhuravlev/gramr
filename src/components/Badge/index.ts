export * from './Badge.types';

export { default } from './Badge';
