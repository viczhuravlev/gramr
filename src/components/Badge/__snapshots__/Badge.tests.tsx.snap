// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] Badge [Snapshot] Badge 1`] = `
<span
  className="s1qvlahm b1t43xoe t1hb7lyd"
  data-size="m"
  style={
    Object {
      "--t1hb7lyd-0": "400",
      "--t1hb7lyd-1": "",
      "--t1hb7lyd-2": "inherit",
    }
  }
>
  Steve fish muskellunge swallower triplefin blenny horsefish wolffish.
</span>
`;
